src/lib/generate-site-name.ts

```diff
@@ -26,22 +27,30 @@ export function generateSiteName( usedSites: SiteDetails[] ): string {
 
 	const defaultName = __( 'My WordPress Website' );
 
-	const isPathUnique = ( name: string ): boolean => {
+	const isPathUnique = async ( name: string ): Promise< boolean > => {
 		const sanitizedPath = sanitizeFolderName( name );
-		return ! usedSites.some( ( site ) => site.path.toLowerCase().endsWith( '/' + sanitizedPath ) );
+		for ( const site of usedSites ) {
+			if ( await getIpcApi().isPathEndingWith( site.path, `/${ sanitizedPath }` ) ) {
+				return false;
+			}
+		}
+		return true;
```

Comment:
```tsx
		const { isEmpty } = await getIpcApi().generateProposedSitePath( name );
		return isEmpty;
```
Could we do something like this instead? It seems like we're kind of duplicating logic now between `generateProposedSitePath` and `isPathEndingWith` now, and this solution seems to work well in my testing.

This would also allow us to remove `isPathEndingWith` altogether, simplifying this PR.